core/base/src/TApplication.cxx

```diff
@@ -628,21 +630,416 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+   // inlineCheck is not zero only if the function is defined as inline.
+   if (inlineCheck == 0) {
+      // If the function is not inline we only change the spacing in "returnType".
+      returnType.ReplaceAll("*", " *");
+   } else {
+      // We check if the function is defined as virtual.
+      Int_t virtualCheck = (func->Property() & kIsVirtual);
+      if (virtualCheck != 0) {
+         // If the function is virtual, we append "virtual" before the return type.
+         TString buffer("virtual ");
+         buffer.Append(returnType);
+         returnType.Clear();
+         returnType.Append(buffer);
+      }
+      returnType.ReplaceAll(" *", "*");
+   }
+   // In any case (with no respect to virtual/inline check) we need to change
+   // the return type as following.
+   returnType.ReplaceAll("istream", "std::istream");
+   returnType.ReplaceAll("ostream", "std::ostream");
+   returnType.ReplaceAll("&", " &");
+   return returnType;
+}
+}
+
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+/// The function generates in a URL for "enumeratorName" defined in "className".
+/// It returns a TString with the URL used in the online reference guide,
+/// generated with Doxygen. The URL generation is specific. It could be
+/// separated into 3 parts - URL for "className", part for the enumeration
+/// and a part for the enumerator.
+///
+/// \param[in] className the name of the class
+/// \param[in] enumeratorName the name of the enumerator from the class
+/// \param[in] dataMember pointer to the enumerator
+/// \param[in] scopeType enumerator to the scope type
 
-   Printf("\nROOT special commands.");
-   Printf("===========================================================================");
-   Printf("             pwd          : show current directory, pad and style");
-   Printf("             ls           : list contents of current directory");
-   Printf("             which [file] : shows path of macro file");
+static TString URLforEnumerator(TString &className, const TString &enumeratorName, TDataMember *dataMember, EUrl scopeType)
+{
+   // We create a TString with the name of the class and the enumeration from which "enumeratorName" is.
+   TString classEnumeration = dataMember->GetTrueTypeName();
+   TString md5EnumClass;
+   // If we have anonymous enumeration we need other syntax ("className::@1@1").
+   if ((classEnumeration.Contains("(anonymous)"))) {
+      // FIXME: need to invetigate the numbering scheme.
+      md5EnumClass.Append(className);
+      md5EnumClass.Append("::@1@1");
+   } else {
+      // If the enumeration is not anonymous we put "className::Enumeration" in a TString,
+      // which will be crypted with MD5 later.
+      md5EnumClass.Append(classEnumeration);
+      TString delim("::");
+      // We extract the part after "::" (this is the enumerator name).
+      TObjArray enumOnly(*classEnumeration.Tokenize(delim));
```

Comment:
Who owns the `TObjString`s returned by Tokenize? Who owns the `TObjArray` - who deletes it? @couet could you see with @dnkalaydjieva to maybe look for the final `':'` instead of tokenizing the string?